versioned_hdf5/subchunk_map.py

```diff
@@ -36,280 +52,405 @@
     )
 
 
-@cython.cfunc
-def _subindex_chunk_slice(
-    s_start: Py_ssize_t,
-    s_stop: Py_ssize_t,
-    s_step: Py_ssize_t,
-    c_start: Py_ssize_t,
-    c_stop: Py_ssize_t,
-) -> slice:
-    """Given a slice(s_start, s_stop, s_step) indexing an axis of
-    a dataset, return the slice of the output array (on __getitem__)
-    or value parameter array (on __setitem__) along the
-    same axis that is targeted by the same slice after it's been
-    clipped to select only the data within the range [c_start, c_stop[
-
-    In other words:
-
-    a = _subindex_chunk_slice(s_start, s_stop, s_step, c_start, c_stop)
-    b = _subindex_slice_chunk(s_start, s_stop, s_step, c_start, c_stop)
-
-    __getitem__: out[a] = cache_chunk[b]
-    __setitem__: cache_chunk[b] = value[a]
-    """
-    start: Py_ssize_t = max(c_start, s_start)
-    # Get the smallest lcm multiple of common that is >= start
-    start = smallest_step_after(start, s_start % s_step, s_step)
-    # Finally, we need to shift start so that it is relative to index
-    start = (start - s_start) // s_step
+class DropAxis(enum.Enum):
+    _drop_axis = 0
```

Comment:
replaces `()` for scalar indices, for the sake of code readability. It took me a long time to realise that the `()` is never actually used to index an ndarray as it is discarded at the end of `as_subchunk_map`.